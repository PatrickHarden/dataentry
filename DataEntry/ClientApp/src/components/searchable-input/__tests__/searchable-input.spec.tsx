import React from 'react';
import { mount, shallow } from 'enzyme';

import SearchableInput from '../searchable-input';

describe('components', () => {

  describe('searchable-input', () => {

    let props:any;
    let wrapper:any;


    beforeEach(() => {
        props = {
            name: "searchable",
            label: "Search Me",
            defaultValue: "Some Value",
            placeholder: "Search...",
            error: false,
            delayMS: 1000
        }
    });

    // it displays a default value as expected
    it("default value displays as expected", () => {
        wrapper = mount(<SearchableInput {...props}/>);
        expect(wrapper.find('input').instance().value).toEqual("Some Value");
    });

    // it sets the default label as expected
    it("renders the expected label", () => {
      wrapper = mount(<SearchableInput {...props}/>);
      expect(wrapper.find('h5').text()).toEqual("Search Me");
    });

    // snapshot render
    it('renders according to snapshot', () => {
        expect(shallow(<SearchableInput {...props} />)).toMatchSnapshot();
    });
  });
});