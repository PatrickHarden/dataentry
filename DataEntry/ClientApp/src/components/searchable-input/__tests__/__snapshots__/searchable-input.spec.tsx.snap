// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components searchable-input renders according to snapshot 1`] = `
<styled.div
  id="searchable_cont"
  onBlur={[Function]}
>
  <FormFieldLabel
    error={false}
    label="Search Me"
  />
  <styled.div
    error={false}
    tabIndex={-1}
  >
    <Autosuggest
      alwaysRenderSuggestions={false}
      focusInputOnSuggestionClick={true}
      getSuggestionValue={[Function]}
      highlightFirstSuggestion={false}
      id="1"
      inputProps={
        Object {
          "onChange": [Function],
          "placeholder": "Search...",
          "value": "Some Value",
        }
      }
      multiSection={false}
      onSuggestionSelected={[Function]}
      onSuggestionsClearRequested={[Function]}
      onSuggestionsFetchRequested={[Function]}
      renderInputComponent={[Function]}
      renderSuggestion={[Function]}
      renderSuggestionsContainer={[Function]}
      shouldRenderSuggestions={[Function]}
      suggestions={Array []}
      theme={
        Object {
          "container": "react-autosuggest__container",
          "containerOpen": "react-autosuggest__container--open",
          "input": "react-autosuggest__input",
          "inputFocused": "react-autosuggest__input--focused",
          "inputOpen": "react-autosuggest__input--open",
          "sectionContainer": "react-autosuggest__section-container",
          "sectionContainerFirst": "react-autosuggest__section-container--first",
          "sectionTitle": "react-autosuggest__section-title",
          "suggestion": "react-autosuggest__suggestion",
          "suggestionFirst": "react-autosuggest__suggestion--first",
          "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
          "suggestionsContainer": "react-autosuggest__suggestions-container",
          "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
          "suggestionsList": "react-autosuggest__suggestions-list",
        }
      }
    />
    <NoDataDropdown />
  </styled.div>
</styled.div>
`;
